File: Claude-T2/apps/web/src/api/role.ts
import { request } from './request';

export interface Role {
  id: string;
  name: string;
  code: string;
  description?: string;
  permissions?: Array<{ action: string; subject: string }>;
  createdAt?: string;
  updatedAt?: string;
}

/**
 * 获取角色列表
 */
export function getRoles(): Promise<Role[]> {
  return request.get('/roles');
}

/**
 * 获取角色详情
 */
export function getRole(id: string): Promise<Role> {
  return request.get(`/roles/${id}`);
}

File: Claude-T2/apps/api-java/src/main/java/com/lifeline/modules/role/entity/Role.java
package com.lifeline.modules.role.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;

@Getter
@Setter
@Entity
@Table(name = "roles")
public class Role {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(updatable = false, nullable = false)
    private String id;

    @Column(nullable = false, length = 50, unique = true)
    private String name;

    @Column(nullable = false, length = 50, unique = true)
    private String code;

    @Column(length = 200)
    private String description;

    @Column(columnDefinition = "jsonb")
    private String[] permissions;

    @CreatedDate
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @LastModifiedDate
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}
